Source/Core/Core/Analytics.h

```diff
@@ -63,6 +77,22 @@ class DolphinAnalytics
   // values created by MakeUniqueId.
   std::string m_unique_id;
 
+  // Performance sampling configuration constants.
+  //
+  // 5min after startup + rand(0, 3min) jitter time, collect performance for 100 frames in a row.
+  // Repeat collection after 30min + rand(0, 3min).
+  static constexpr int NUM_PERFORMANCE_SAMPLES_PER_REPORT = 100;
+  static constexpr int PERFORMANCE_SAMPLING_INITIAL_WAIT_TIME_SECS = 300;
+  static constexpr int PERFORMANCE_SAMPLING_WAIT_TIME_JITTER_SECS = 180;
+  static constexpr int PERFORMANCE_SAMPLING_INTERVAL_SECS = 1800;
+
+  // Performance sampling state & internal helpers.
+  void InitializePerformanceSampling();  // Called on game start / title switch.
+  bool ShouldStartPerformanceSampling();
+  u64 m_sampling_next_start_us;              // Next timestamp (in us) at which to trigger sampling.
```

Comment:
Don't know how well the interaction would be, but perhaps it would be preferable to use `std::chrono::microseconds` here to enforce the time unit. Though, this would likely need a change to the Timer interface to not look awkward, so I'm totally fine if it's left out.

Reply:
Yeah, I thought about doing it too while writing the code but it ended up making things less readable since our common libs don't support std::chrono... Maybe one day :)